server/service/queries.go

```diff
@@ -405,6 +414,17 @@ func (svc *Service) ModifyQuery(ctx context.Context, id uint, p fleet.QueryPaylo
 	return query, nil
 }
 
+func comparePlatforms(platform1, platform2 string) bool {
```

Comment:
nit: i personally prefer boolean function names in the form of a question ie. `!platformsMatch()` or `!platformsEqual()`

Reply:
Agree. I'll try to rename it when I work on fixing a similar thing on policies (we also need to remove policy membership when platform is changed on the policy)